Slamdunk

t-neumann opened the pull request:
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [ x ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [ x ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ x ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

 * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
 * `@BiocondaBot please add label` will add the `please review & merge` label.
 * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.

For members of the Bioconda project, the following command is also available:

 * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.

If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

t-neumann commented:
I completely refactored the python package so that it is now purely python and pulls in all dependencies from conda directly (nextgenmap, samtools, varscan). That is why I moved the old `meta.yaml` and `build.sh` in a subfolder with version number. But I do not know if this is the way to go, so I would kindly request some advice here how to proceed.

dpryan79 commented:
I'd suggest just deleting the old recipe. We can add it back if someone really needs it rebuilt.

t-neumann commented:
@dpryan79 I'm a bit lost with this error:

```
11:19:27 BIOCONDA INFO (OUT) conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
11:19:27 BIOCONDA INFO (OUT)   - pandas=0.25.3 -> python[version='>=3.8,<3.9.0a0']
11:19:27 BIOCONDA INFO (OUT)   - python=2.7
```

I copied all dependencies + version numbers from an actual running conda environment from my machine where I managed to install a running version of `slamdunk v0.4.0` into. And I checked everything and cannot find any dependency that I fixed for `python2.7`.

Any clue in which direction I should look?

bgruening commented:
That probably means that 0.25.3 is not available for python38. Can you relax the pinning?

t-neumann commented:
@bgruening But correct me if I'm wrong - I definitely see a `tar.bz2` files for `pandas 0.25.3` for `py38` here https://anaconda.org/anaconda/pandas/files

t-neumann commented:
@bgruening Ah but there's no `py2.7`. So should I restrict my build to work only for `py3` since `py2.7` will be obsolete by the end of the year anyways?

dpryan79 commented:
Correct, pandas has dropped support for python 2.7.

bgruening commented:
I would do so, let's drop py27 :)

t-neumann commented:
@dpryan79 I came across a very non-telling error. At first I thought it is compiling stuff, so I added cython again and removed `noarch`, but the error keeps persisting.

```
13:17:38 BIOCONDA INFO (OUT) /opt/conda/conda-bld/slamdunk_1576501934669/work/conda_build.sh: line 6: {{: command not found
13:17:40 BIOCONDA INFO (OUT) Traceback (most recent call last):
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/bin/conda-build", line 11, in <module>
13:17:40 BIOCONDA INFO (OUT)     sys.exit(main())
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 445, in main
13:17:40 BIOCONDA INFO (OUT)     execute(sys.argv[1:])
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 436, in execute
13:17:40 BIOCONDA INFO (OUT)     verify=args.verify, variants=args.variants)
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
13:17:40 BIOCONDA INFO (OUT)     notest=notest, need_source_download=need_source_download, variants=variants)
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2343, in build_tree
13:17:40 BIOCONDA INFO (OUT)     notest=notest,
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1491, in build
13:17:40 BIOCONDA INFO (OUT)     cwd=src_dir, stats=build_stats)
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 397, in check_call_env
13:17:40 BIOCONDA INFO (OUT)     return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
13:17:40 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 377, in _func_defaulting_env_to_os_environ
13:17:40 BIOCONDA INFO (OUT)     raise subprocess.CalledProcessError(proc.returncode, _args)
13:17:40 BIOCONDA INFO (OUT) subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/opt/conda/conda-bld/slamdunk_1576501934669/work/conda_build.sh']' returned non-zero exit status 127.
```

Any ideas?

dpryan79 commented:
I've fixed that, you were using jinja templating stuff in build.sh. This is now pure python, yes?

t-neumann commented:
Gosh yes - I just stupidly copied that from the `meta.yml` and forgot replacing the jinja with `$PYTHON`. Thanks.

dpryan79 commented:
@bioconda-bot please fetch artifacts

dpryan79 commented on recipes/slamdunk/meta.yaml:
```diff
@@ -14,23 +14,27 @@ build:
+    - pandas=0.25.3
+    - biopython=1.74
+    - pybedtools=0.8.0
+    - intervaltree=3.0.2
```
Do we really need such exact pinning? For example, will samtools 1.9 really not work? This seems overly strict.

t-neumann replied:
I just created a conda env for testing and wrote that to an `environment.yml`. I kinda need it reproducible though (at least with regards to `nextgenmap` and `varscan`), so samtools we could relax to `samtools>=1.9` for sure.

dpryan79 replied:
What about pandas, biopython and the other python dependencies?

dpryan79 replied:
Basically, the more packages that we have that are strictly pinned as dependencies the more likely the recipe as a whole will become unmaintainable quickly.

t-neumann replied:
Those should be fine more relaxed as well - again just pulled out of the conda environment.

dpryan79 commented:
@t-neumann If you're happy with the recipe as it is now we can merge.

t-neumann commented:
Looks good! go ahead!
